File: app/src/main/java/ua/regin/badproduct/manager/impl/AdditiveManager.java
package ua.regin.badproduct.manager.impl;

import com.firebase.client.Firebase;

import javax.inject.Inject;

import ua.regin.badproduct.manager.IAdditiveManager;

public class AdditiveManager implements IAdditiveManager {

    private final Firebase firebase;

    @Inject
    public AdditiveManager(Firebase firebase) {
        this.firebase = firebase;
    }

    @Override
    public void additiveListener() {

    }
}

File: app/src/main/java/ua/regin/badproduct/manager/impl/AdditiveManager.kt
package ua.regin.badproduct.manager.impl

import android.content.Context
import com.firebase.client.Firebase
import com.firebase.client.ValueEventListener
import org.androidannotations.annotations.EBean
import ua.regin.badproduct.application.Application
import ua.regin.badproduct.manager.IAdditiveManager

@EBean(scope = EBean.Scope.Singleton)
open class AdditiveManager : IAdditiveManager {

    private val firebase: Firebase;
    private val context: Context;

    constructor(context: Context) {
        this.context = context;
        firebase = Firebase(Application.FIREBASE_URL);
    }

    override fun addAdditiveListener(valueEventListener: ValueEventListener) {
        firebase.addValueEventListener(valueEventListener)
    }

    override fun removeAdditiveListener(valueEventListener: ValueEventListener) {
        firebase.removeEventListener(valueEventListener)
    }
}
